### ISSUE TITLE
Virtualbox build broken

### ISSUE THREAD

**Issue by tokudan** [NixOS/nixpkgs#16260]
## Issue description
I'm unable to build virtualbox through nixos-rebuild (error messages at the end)

### Steps to reproduce
configuration.nix:
  virtualisation.virtualbox.host.enable = true;

nixos-rebuild test

## Technical details

* System: 16.03.909.f199be0 (Emu)
* Nix version: nix-env (Nix) 1.11.2
* Nixpkgs version: "16.03.923.9e959ce"

---
```
  CC [M]  /tmp/nix-build-virtualbox-5.0.12-4.6.2.drv-0/VirtualBox-5.0.12/out/linux.amd64/release/bin/src/vboxdrv/r0drv/linux/memobj-r0drv-linux.o
/tmp/nix-build-virtualbox-5.0.12-4.6.2.drv-0/VirtualBox-5.0.12/out/linux.amd64/release/bin/src/vboxdrv/r0drv/linux/memobj-r0drv-linux.c: In function 'rtR0MemObjNativeFree':
/tmp/nix-build-virtualbox-5.0.12-4.6.2.drv-0/VirtualBox-5.0.12/out/linux.amd64/release/bin/src/vboxdrv/r0drv/linux/memobj-r0drv-linux.c:581:21: error: implicit declaration of function 'page_cache_release' [-Werror=implicit-function-declaration]
                     page_cache_release(pMemLnx->apPages[iPage]);
                     ^
/tmp/nix-build-virtualbox-5.0.12-4.6.2.drv-0/VirtualBox-5.0.12/out/linux.amd64/release/bin/src/vboxdrv/r0drv/linux/memobj-r0drv-linux.c: In function 'rtR0MemObjNativeLockUser':
/tmp/nix-build-virtualbox-5.0.12-4.6.2.drv-0/VirtualBox-5.0.12/out/linux.amd64/release/bin/src/vboxdrv/r0drv/linux/memobj-r0drv-linux.c:1032:29: warning: passing argument 1 of 'get_user_pages' makes integer from pointer without a cast [-Wint-conversion]
         rc = get_user_pages(pTask,                  /* Task for fault accounting. */
                             ^
In file included from /tmp/nix-build-virtualbox-5.0.12-4.6.2.drv-0/VirtualBox-5.0.12/out/linux.amd64/release/bin/src/vboxdrv/r0drv/linux/the-linux-kernel.h:88:0,
                 from /tmp/nix-build-virtualbox-5.0.12-4.6.2.drv-0/VirtualBox-5.0.12/out/linux.amd64/release/bin/src/vboxdrv/r0drv/linux/memobj-r0drv-linux.c:31:
/nix/store/rlcs9qvclsxavc0kz305gyv1fcq1i8s7-linux-4.6.2-dev/lib/modules/4.6.2/source/include/linux/mm.h:1266:6: note: expected 'long unsigned int' but argument is of type 'struct task_struct *'
 long get_user_pages(unsigned long start, unsigned long nr_pages,
      ^
/tmp/nix-build-virtualbox-5.0.12-4.6.2.drv-0/VirtualBox-5.0.12/out/linux.amd64/release/bin/src/vboxdrv/r0drv/linux/memobj-r0drv-linux.c:1033:29: warning: passing argument 2 of 'get_user_pages' makes integer from pointer without a cast [-Wint-conversion]
                             pTask->mm,              /* Whose pages. */
                             ^
In file included from /tmp/nix-build-virtualbox-5.0.12-4.6.2.drv-0/VirtualBox-5.0.12/out/linux.amd64/release/bin/src/vboxdrv/r0drv/linux/the-linux-kernel.h:88:0,
                 from /tmp/nix-build-virtualbox-5.0.12-4.6.2.drv-0/VirtualBox-5.0.12/out/linux.amd64/release/bin/src/vboxdrv/r0drv/linux/memobj-r0drv-linux.c:31:
/nix/store/rlcs9qvclsxavc0kz305gyv1fcq1i8s7-linux-4.6.2-dev/lib/modules/4.6.2/source/include/linux/mm.h:1266:6: note: expected 'long unsigned int' but argument is of type 'struct mm_struct *'
 long get_user_pages(unsigned long start, unsigned long nr_pages,
      ^
/tmp/nix-build-virtualbox-5.0.12-4.6.2.drv-0/VirtualBox-5.0.12/out/linux.amd64/release/bin/src/vboxdrv/r0drv/linux/memobj-r0drv-linux.c:1036:29: warning: passing argument 5 of 'get_user_pages' makes pointer from integer without a cast [-Wint-conversion]
                             fWrite,                 /* Write to memory. */
                             ^
In file included from /tmp/nix-build-virtualbox-5.0.12-4.6.2.drv-0/VirtualBox-5.0.12/out/linux.amd64/release/bin/src/vboxdrv/r0drv/linux/the-linux-kernel.h:88:0,
                 from /tmp/nix-build-virtualbox-5.0.12-4.6.2.drv-0/VirtualBox-5.0.12/out/linux.amd64/release/bin/src/vboxdrv/r0drv/linux/memobj-r0drv-linux.c:31:
/nix/store/rlcs9qvclsxavc0kz305gyv1fcq1i8s7-linux-4.6.2-dev/lib/modules/4.6.2/source/include/linux/mm.h:1266:6: note: expected 'struct page **' but argument is of type 'int'
 long get_user_pages(unsigned long start, unsigned long nr_pages,
      ^
/tmp/nix-build-virtualbox-5.0.12-4.6.2.drv-0/VirtualBox-5.0.12/out/linux.amd64/release/bin/src/vboxdrv/r0drv/linux/memobj-r0drv-linux.c:1037:29: warning: passing argument 6 of 'get_user_pages' makes pointer from integer without a cast [-Wint-conversion]
                             fWrite,                 /* force write access. */
                             ^
In file included from /tmp/nix-build-virtualbox-5.0.12-4.6.2.drv-0/VirtualBox-5.0.12/out/linux.amd64/release/bin/src/vboxdrv/r0drv/linux/the-linux-kernel.h:88:0,
                 from /tmp/nix-build-virtualbox-5.0.12-4.6.2.drv-0/VirtualBox-5.0.12/out/linux.amd64/release/bin/src/vboxdrv/r0drv/linux/memobj-r0drv-linux.c:31:
/nix/store/rlcs9qvclsxavc0kz305gyv1fcq1i8s7-linux-4.6.2-dev/lib/modules/4.6.2/source/include/linux/mm.h:1266:6: note: expected 'struct vm_area_struct **' but argument is of type 'int'
 long get_user_pages(unsigned long start, unsigned long nr_pages,
      ^
/tmp/nix-build-virtualbox-5.0.12-4.6.2.drv-0/VirtualBox-5.0.12/out/linux.amd64/release/bin/src/vboxdrv/r0drv/linux/memobj-r0drv-linux.c:1032:14: error: too many arguments to function 'get_user_pages'
         rc = get_user_pages(pTask,                  /* Task for fault accounting. */
              ^
In file included from /tmp/nix-build-virtualbox-5.0.12-4.6.2.drv-0/VirtualBox-5.0.12/out/linux.amd64/release/bin/src/vboxdrv/r0drv/linux/the-linux-kernel.h:88:0,
                 from /tmp/nix-build-virtualbox-5.0.12-4.6.2.drv-0/VirtualBox-5.0.12/out/linux.amd64/release/bin/src/vboxdrv/r0drv/linux/memobj-r0drv-linux.c:31:
/nix/store/rlcs9qvclsxavc0kz305gyv1fcq1i8s7-linux-4.6.2-dev/lib/modules/4.6.2/source/include/linux/mm.h:1266:6: note: declared here
 long get_user_pages(unsigned long start, unsigned long nr_pages,
      ^
cc1: some warnings being treated as errors
/nix/store/rlcs9qvclsxavc0kz305gyv1fcq1i8s7-linux-4.6.2-dev/lib/modules/4.6.2/source/scripts/Makefile.build:291: recipe for target '/tmp/nix-build-virtualbox-5.0.12-4.6.2.drv-0/VirtualBox-5.0.12/out/linux.amd64/release/bin/src/vboxdrv/r0drv/linux/memobj-r0drv-linux.o' failed
make[3]: *** [/tmp/nix-build-virtualbox-5.0.12-4.6.2.drv-0/VirtualBox-5.0.12/out/linux.amd64/release/bin/src/vboxdrv/r0drv/linux/memobj-r0drv-linux.o] Error 1
/nix/store/rlcs9qvclsxavc0kz305gyv1fcq1i8s7-linux-4.6.2-dev/lib/modules/4.6.2/source/Makefile:1429: recipe for target '_module_/tmp/nix-build-virtualbox-5.0.12-4.6.2.drv-0/VirtualBox-5.0.12/out/linux.amd64/release/bin/src/vboxdrv' failed
make[2]: *** [_module_/tmp/nix-build-virtualbox-5.0.12-4.6.2.drv-0/VirtualBox-5.0.12/out/linux.amd64/release/bin/src/vboxdrv] Error 2
Makefile:146: recipe for target 'sub-make' failed
make[1]: *** [sub-make] Error 2
Makefile:24: recipe for target '__sub-make' failed
make: *** [__sub-make] Error 2
make: Leaving directory '/nix/store/rlcs9qvclsxavc0kz305gyv1fcq1i8s7-linux-4.6.2-dev/lib/modules/4.6.2/build'
builder for ‘/nix/store/s5wxzl935fc06y1b5hr8zpbgc6a8mkcd-virtualbox-5.0.12-4.6.2.drv’ failed with exit code 2
cannot build derivation ‘/nix/store/g77mjfblc9qgkqx0xk01kmyciq42l0qh-nixos-system-yuna-16.03.923.9e959ce.drv’: 1 dependencies couldn't be built
error: build of ‘/nix/store/g77mjfblc9qgkqx0xk01kmyciq42l0qh-nixos-system-yuna-16.03.923.9e959ce.drv’ failed
```

**Comment by bjornfor** [comment]
(Your configuration.nix specifies a newer Linux version than what NixOS 16.03 uses as default, but that was not mentioned in the issue text.)

I believe this was fixed in commit 76ebea5f534b7a3cf4fab61d6d8cb4c3acae4080, which will hit the channels soon.

**Comment by tokudan** [comment]
Yep, it's working now.
